search/lib/api/provider.js

```diff
@@ -50,6 +50,86 @@ async function getProvider (request, response) {
   }
 }
 
+/**
+ * Get all the cloud holding collections for the provider in the request.
+ */
+async function getCloudProvider (request, response) {
+  try {
+    const providerId = request.params.providerId;
+    logger.info(`GET /${providerId}`);
+    const event = request.apiGateway.event;
+
+    // validate that providerId is valid
+    const providerList = await cmr.getProviderList();
+    const isProvider = providerList.filter(providerObj => providerObj['provider-id'] === providerId);
+    if (isProvider.length === 0) throw new Error(`Provider [${providerId}] not found`);
+
+    //Query params to get cloud holdings for the provider.
+    const params = Object.assign(
+      { provider_short_name: providerId },
+      { tag_key: "gov.nasa.earthdatacloud.s3"},
+      //Used for pagination.
+      //await cmr.convertParams(provider, request.query)
+      { page_num: request.query.page}
```

Comment:
What happens if additional query parameters other than page are passed in? Are they just ignored?  I'm not actually sure what best practices are for REST APIs, should an error be thrown?

Also, if page is accepted, it usually makes sense to accept `limit` (i.e., page size), but not required.

Reply:
Per our discussion, I will use the cmr.convertParams that's commented out.